### DISCUSSION TITLE
Unexpected behaviour for Django models definitions

### DISCUSSION THREAD

**Discussion by brdacost** [microsoft/pyright#5885] [General]
I have the following model definition on my Django project

```
from django.db import models
from django.utils.translation import gettext_lazy as _


class BaseModel(models.Model):
    created_at = models.DateTimeField(_("Created At"), auto_now_add=True)
    last_updated_at = models.DateTimeField(_("Last Updated at"), auto_now=True)
    deleted = models.BooleanField(_("Deleted"), default=False)

    class Meta:
        abstract = True

```

The code works exactly as expected, but the type hecking does not, beause it keeps acusing errors like

```
Pyright: Argument of type "Literal[False]" cannot be assigned to parameter "default" of type "type[NOT_PROVIDED]" in function "__init__"
  Type "Literal[False]" cannot be assigned to type "type[NOT_PROVIDED]" [reportGeneralTypeIssues]

```
It seems like the django types are not well definied and a lot of other annoying errors seem to follow a similar logic, for example, when defining a charfield in a model and attributing a string value to a model instance, pyright is also throwing up some errors like

```
class State(BaseModel):
    name = models.CharField(_("Name"), max_length=50, blank=False, null=False)
    abbreviation = models.CharField(
        _("Abbreviation"), max_length=2, blank=False, null=False
    )

state_mt = State()
state_mt.name = "Mato Grosso"
>>> "Literal['Mato Grosso']" is incompatible with "CharField" [reportGeneralTypeIssues]
state_mt.abbreviation = "MT"
>>> "Literal['MT']" is incompatible with "CharField" [reportGeneralTypeIssues]
```

What is the bewt way to fix and or workaround this problem?

**Comment by erictraut** [comment]
I'm not able to repro any type checking errors with your first code sample. 

Which version of django do you have installed? I've installed version 4.2.5, which is the latest published version.

Do you have the `django-stubs` package installed as well? If so, which version? I have version 4.2.3 installed, although I've also tried to repro your problem without the stubs installed. Django is an untyped library (i.e. it doesn't have a "py.typed" marker file that indicates it has bundled type information), so installing the separate stub library is recommended if you want to use it with static type checkers.
